src/System.Private.ServiceModel/tests/Scenarios/Contract/Fault/FaultExceptionTests.4.0.0.cs

```diff
@@ -299,4 +301,74 @@ public static void FaultException_FaultContractAndKnownType_Echo()
             ScenarioTestHelpers.CloseCommunicationObjects((ICommunicationObject)serviceProxy, factory);
         }
     }
+
+#if FULLXUNIT_NOTSUPPORTED
+    [Fact]
+#endif
+    [WcfFact]
+    [OuterLoop]
+    public static void TestAccessFaultContractInfo()
+    {
+        MyClientBase<IWcfService> client = null;
+        IWcfService serviceProxy = null;
+
+        try
+        {
+            // *** SETUP *** \\
+            CustomBinding customBinding = new CustomBinding();
+            customBinding.Elements.Add(new TextMessageEncodingBindingElement());
+            customBinding.Elements.Add(new HttpTransportBindingElement());
+
+            client = new MyClientBase<IWcfService>(customBinding, new EndpointAddress(Endpoints.DefaultCustomHttp_Address));
+            client.Endpoint.EndpointBehaviors.Add(new TestFaultContractInfosBehavior());
+            serviceProxy = client.ChannelFactory.CreateChannel();
+
+            // *** EXECUTE *** \\
+            var input = new object[] { new FaultDetail(), new KnownTypeA() };
+            // Use exiting service as we only need to verify FaultContractInfo can be accessed from ClientOperation
+            serviceProxy.TestFaultWithKnownType(null, input);
+        }
+        finally
+        {
+            // *** ENSURE CLEANUP *** \\
+            ScenarioTestHelpers.CloseCommunicationObjects((ICommunicationObject)serviceProxy, (ICommunicationObject)client);
+        }
+    }
 }
+
+public class TestFaultContractInfosBehavior : IEndpointBehavior
```

Comment:
What I was thinking was this -- add a property to this TestFaultContractInfosBehavior class that returns the ClientRuntime you get in ApplyClientBehavior.  Then in your test, keep a reference to the TestFaultContractInfosBehavior you create.  In the test, you can refer to the ClientRuntime property that you captured and do the same checks you do here -- but inside the test method instead of here.